src/StepAdaptationController/src/StepAdaptationController.cpp

```diff
@@ -487,9 +575,397 @@ bool StepAdaptationController::getAdaptatedFootTrajectory(const footTrajectoryGe
     return true;
 }
 
+bool StepAdaptationController::triggerStepAdapterByArmCompliant(const double &numberOfActuatedDof, const iDynTree::VectorDynSize &qDesired, const iDynTree::VectorDynSize &qActual,
+                                                                const std::deque<bool>& leftInContact,const std::deque<bool>& rightInContact,std::vector<std::string> jointsListVector)
+{
+    double leftArmPitchError=0;
+    double rightArmPitchError=0;
+    double leftArmRollError=0;
+    double rightArmRollError=0;
+
+    for (int var=0;var<m_pushDetectionListRightArmX.size();++var)
+    {
+        std::vector<std::string>::iterator it = std::find(jointsListVector.begin(), jointsListVector.end(), m_pushDetectionListRightArmX.at(var));
```

Comment:
``find`` may require some time. Maybe you can store the corresponding indexes at configuration time.

Reply:
If I remember correctly the philosophy of these changes was that if someone add or remove some joints to joints list it does not cause the problem. So By putting the indexes inside the configuration file we assuming an specific joints list. The run time of the whole  triggerStepAdapterByArmCompliant function is 3000 nanoseconds that is 0.003 miliseconds.

Reply:
I meant to save the indexes once at the beginning in the ``configure`` method (or similar). But if it is so fast, then fine.

Reply:
Anyhow, notice that this is supposed to run on the robot head, which may not be as powerful as your pc.

Reply:
Thanks. Generally the array does not include a lot of members and I don't think the `find` will not be a problem.